### ISSUE TITLE
'export V=$(echo "{ 'ABC': 123 }")' is not reported as an error

### ISSUE THREAD

**Issue by redbaron** [koalaman/shellcheck#595]
I am not sure if it is something should be reported as bug, I am not a POSIX expert,but  following code is failing in `dash` and `ksh`, but not in `bash`  in posix mode

```sh
export V=$(echo "{ 'ABC': 123 }")
/bin//dash: 1: export: 'ABC':: bad variable name
```

presence of export is important, without it it works. So maybe shellcheck should be reporting it too.

**Comment by bittorf** [comment]
when pasting this into shellcheck.net it complains about SC2155 and SC2116.
it seems to work with dash when you use quotes, e.g.
```
#!/bin/dash
export V="$( echo "{ 'ABC': 123 }" )"
```

**Comment by bittorf** [comment]
also 'export' is a builtin, so how can it be missing?

**Comment by kurahaupo** [comment]
Where the shell recognizes a variable assignment as part of syntax, it's
treated a little specially; in particular the RHS is not subject to globbing
or word-splitting.

The issue is that some special built-ins can be followed by assignments that
get the special treatment, whereas "ordinary" built-ins are just text and so
what are intended as assignments get subject to wordsplitting and globbing
before they're recognized as assignments.

And which builtins are "special" in this way varies from shell to shell.
POSIX mandates a particular list, and unfortunately both omitting specified
words and adding extra words would make an implementation non-conforming.

**Comment by kurahaupo** [comment]
On Fri, 5 Feb 2016, Bastian Bittorf wrote:
> also 'export' is a builtin, so how can it be missing?

The recognition of assignments does not apply to all builtins; it does not
apply to "cd" for instance.

**Comment by Arthur2e5** [comment]
This is related to a pending POSIX update. See https://lists.gnu.org/archive/html/bug-bash/2015-12/msg00019.html.

**Comment by redbaron** [comment]
until new POSIX standard is implemented in popular shells, I think it would be useful if shellcheck warned about it, do you agree?

**Comment by Arthur2e5** [comment]
Good to warn, just providing some extra info (and perhaps an excuse to say 'oh, tomorrow' forever).
